pubtools/_content_gateway/push_base.py

```diff
@@ -0,0 +1,463 @@
+            - Search for the file record from the file mapping (file_mapping) mapping
+            - If `file state` == `present` and record not found in mapping
+
+                - Creates a new file record associated with product on CGW with given data
+                - Add the new record into the file mapping records
+
+            - If `file state` == `present` and record found in mapping
+
+                - Update the file record on content gateway with given data
+                - Update the record into the file mapping records
+
+            - If `file state` == `absent` and record found in mapping
+
+                - Delete the file record on the content gateway
+                - Remove the record from the file mapping records
+
+            - If `file state` == `absent` and record not found in mapping
+
+                - Raises CGWError error
+
+        Args:
+            file_item (list(dict)):
+                List of file record
+
+        Raises:
+            CGWError:
+                When file record is not present and tries to delete the same
+
+        Returns:
+            file_id (int|None):
+                Returns the file ID
+        """
+
+        LOG.debug("Fetching file id for the file metadata")
+        product_name = file_item.get('metadata').get('productName')
+        product_code = file_item.get('metadata').get('productCode')
+        version_name = file_item.get('metadata').get('productVersionName')
+        download_url = file_item.get('metadata').get('downloadURL')
+        product_id = self.product_mapping.get((product_name, product_code))
+
+        if not product_id:
+            raise CGWError("Product name: %s product code: %s not found" % (product_name, product_code))
+
+        version_id = self.pv_mapping.get((product_name, product_code, version_name))
+        if not version_id:
+            raise CGWError("Product name: %s product code: %s version name: %s not found" % (
+                product_name, product_code, version_name))
+
+        file_id = self.file_mapping.get((product_name, product_code, version_name, download_url))
+
+        if file_item.get('state') == 'present':
+            del file_item.get('metadata')['productName'], \
+                file_item.get('metadata')['productCode'], \
+                file_item.get('metadata')['productVersionName']
+            file_item.get('metadata')['productVersionId'] = version_id
+
+            if not file_id:
+                LOG.info("No previous entries found for the given file metadata")
+                LOG.info("Creating file entry for the given downloadURL: %s ", download_url)
+                file_id = self.cgw_client.create_file(product_id, version_id, file_item.get('metadata'))
+                self.file_mapping[
+                    (product_name, product_code, version_name, download_url)] = file_id
+                LOG.info("New file created with file_id:- %s" % file_id)
+                return
```

Comment:
We don't want to return file id?

Reply:
It is optional. We only need the id in case of delete operation for the test case script.